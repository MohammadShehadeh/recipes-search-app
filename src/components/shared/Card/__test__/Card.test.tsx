import renderer from 'react-test-renderer';

import { Card } from '..';

const cardMock = {
  image: 'image',
  id: 10,
  title: 'title',
};

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  Link: ({ children }: any) => <a data-mocked-link>{children}</a>,
}));

describe('Card', () => {
  it('should render to match snapshot', () => {
    const tree = renderer.create(<Card {...cardMock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
