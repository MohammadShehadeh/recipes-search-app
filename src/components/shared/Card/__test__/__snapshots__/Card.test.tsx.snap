// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render to match snapshot 1`] = `
<a
  data-mocked-link={true}
>
  <h3
    className="title"
  >
    title
  </h3>
  <div
    className="image"
  >
    <img
      alt="title"
      src="image"
    />
  </div>
</a>
`;
